src/main/java/io/codelex/flightplanner/Repository/FlightsRepository.java

```diff
@@ -0,0 +1,125 @@
+package io.codelex.flightplanner.Repository;
+
+import io.codelex.flightplanner.Modules.*;
+import org.springframework.http.HttpStatus;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.server.ResponseStatusException;
+
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+
+
+@Repository
+public class FlightsRepository {
+
+    private final List<Flight> flightList = new ArrayList<>();
+    private int idCount = 1;
+
+    public void clearFlight() {
+        flightList.clear();
+    }
+
+    public void deleteFlight(int id) {
+        flightList.removeIf(flight -> flight.getId() == id);
+        throw new ResponseStatusException(HttpStatus.OK);
+    }
+
+    public Flight addFlight(AddFlightRequest flightRequest) {
+
+        Flight flight = new Flight(idCount, flightRequest);
+
+        if (flight.getFrom().equals(flight.getTo()) || !correctDateFormat(flight)) {
+            throw new ResponseStatusException(HttpStatus.BAD_REQUEST);
+
+        } else if (!sameFlightInList(flight)) {
+            idCount++;
+            flightList.add(flight);
+            return flight;
+
+        } else {
+            throw new ResponseStatusException(HttpStatus.CONFLICT);
+        }
+    }
+
+    private boolean correctDateFormat(Flight flight) {
+        return flight.getDepartureTime().isBefore(flight.getArrivalTime());
+    }
+
+    private boolean sameFlightInList(Flight flight) {
+        boolean sameFlight = false;
+        for (Flight thisFlight: flightList) {
+            if (flight.getFrom().equals(thisFlight.getFrom())
+                    && flight.getTo().equals(thisFlight.getTo())
+                    && flight.getCarrier().equals(thisFlight.getCarrier())
+                    && flight.getDepartureTime().equals(thisFlight.getDepartureTime())
+                    && flight.getArrivalTime().equals(thisFlight.getArrivalTime())) {
+                sameFlight = true;
+                break;
+            }
+        }
+        return sameFlight;
+    }
+
+    public Flight fetchFlight(int id) {
+        Flight found = null;
+        for (Flight thisFlight: flightList) {
```

Comment:
arī daudz īsāk ar stream sanāks
in general 99% labāk izmantot stream 👍

Reply:
done.